wagtail/admin/forms/choosers.py

```diff
@@ -76,24 +76,7 @@ def filter(self, objects):
         search_query = self.cleaned_data.get("q")
         if search_query:
             search_backend = get_search_backend()
-
-            # The search should work as an autocomplete by preference, but only if
-            # there are AutocompleteFields set up for this model in the search index;
-            # if not, fall back on a standard search so that we get results at all
-            if objects.model.get_autocomplete_search_fields():
-                try:
-                    objects = search_backend.autocomplete(search_query, objects)
-                except NotImplementedError:
-                    # Older search backends do not implement .autocomplete() but do support
-                    # partial_match on .search(). Newer ones will ignore partial_match.
-                    objects = search_backend.search(
-                        search_query, objects, partial_match=True
-                    )
-            else:
-                objects = search_backend.search(
-                    search_query, objects, partial_match=True
-                )
-
```

Comment:
Nice! This means it should be fairly simple to apply autocomplete to snippets' IndexView. I'll do that after this is merged.